crates/rune-macros/src/function.rs

```diff
@@ -36,6 +37,14 @@ impl FunctionAttrs {
                 out.instance = true;
             } else if ident == "keep" {
                 out.keep = true;
+            } else if ident == "protocol" {
+                input.parse::<Token![=]>()?;
+                let protocol: syn::Path = input.parse()?;
+                out.path = Path::Protocol(if protocol.segments.len() == 1 {
+                    parse_quote_spanned!(protocol.span()=> rune::runtime::Protocol::#protocol)
```

Comment:
:thinking: I do have a preference towards building syn structures manually as much as possible, but I suppose this is fine. I would note that building and manipulating a `syn::Path` by hand is not hard.

Reply:
I saw that. Personally, I don't think the type-safety one gets by that really offsets the convenience of `quote!`, but I can change it.